doc/examples/data/plot_specific.py

```diff
@@ -55,9 +55,52 @@
 
 fig, axes = plt.subplots(4, 5, figsize=(20, 20))
 ax = axes.ravel()
-images = data.lfw_subset()
+lfs_images = data.lfw_subset()
 for i in range(20):
-    ax[i].imshow(images[90+i], cmap=plt.cm.gray)
-    ax[i].axis('off')
+    ax[i].imshow(lfs_images[90 + i], cmap=plt.cm.gray)
+    ax[i].axis("off")
 fig.tight_layout()
 plt.show()
+
+
+######################################################################
+# Thumbnail image for the gallery
+
+# sphinx_gallery_thumbnail_number = -1
+from matplotlib.offsetbox import AnchoredText
+
+ax1 = plt.subplot(2, 2, 1)
```

Comment:
This is the old Matlab-style API. I'd rather recommend Matplotlib's object API here with `fig, axes = plt.subplots(...)`. You could also use [GridSpec](https://matplotlib.org/stable/gallery/subplots_axes_and_figures/gridspec_multicolumn.html#sphx-glr-gallery-subplots-axes-and-figures-gridspec-multicolumn-py) here.

Reply:
<img width="432" alt="image" src="https://user-images.githubusercontent.com/44469195/161379628-f6f85961-6c2e-4796-a9e8-50026da77e33.png">
Ok, i will change it to this syntax then

Reply:
After some tinkering, I don't like the the GridSpace too much. I know the Matlab-style API is discouraged, but the result looks better and the code is easier to read in my opinion.
Nevertheless, here would be a gridspec option to create the thumbnail:
```py
from matplotlib.offsetbox import AnchoredText

fig = plt.figure(constrained_layout=False)
gs = fig.add_gridspec(nrows=3, ncols=4)
ax0 = fig.add_subplot(gs[0:2, 0:2])
ax0.imshow(cylce_images[0])
ax0.add_artist(
    AnchoredText(
        "Stereo",
        prop=dict(size=20),
        frameon=True,
        borderpad=0,
        loc="upper left",
    )
)
ax1 = fig.add_subplot(gs[0:2, 2:4])
ax1.imshow(vortex_images[0])
ax1.add_artist(
    AnchoredText(
        "PIV",
        prop=dict(size=20),
        frameon=True,
        borderpad=0,
        loc="upper left",
    )
)
ax2 = fig.add_subplot(gs[2, 0])
ax2.imshow(lfw_images[90 + 1], cmap="gray")
ax3 = fig.add_subplot(gs[2, 1])
ax3.imshow(lfw_images[90 + 2], cmap="gray")
ax4 = fig.add_subplot(gs[2, 2])
ax4.imshow(lfw_images[90 + 3], cmap="gray")
ax5 = fig.add_subplot(gs[2, 3])
ax5.imshow(lfw_images[90 + 4], cmap="gray")
for ax in [ax0,ax1, ax2, ax3, ax4, ax5]:
    ax.axis("off")
```
<img width="156" alt="image" src="https://user-images.githubusercontent.com/44469195/161380024-b5b671c2-b7ed-4157-9079-46258e2d44a3.png">
vs. current thumbnail:
<img width="156" alt="image" src="https://user-images.githubusercontent.com/44469195/161380082-04173ce5-efc7-40eb-9b01-3cfb2fefaffa.png">
@lagru : Feel free to use this code for some more tweaking if you want to :) The changes you can direclty commit to this branch.